### PR TITLE
refactor(engine): use evaluation.proto as source of truth for deseria…

### PR DESCRIPTION
**Pull request by erka**
…lize Flipt response

### CODE REVIEW

**Review by yquansah — COMMENTED** [review]
This is amazing work @erka 👏! Having one source of truth for the data models seems appropriate.

File: flipt-engine/proto/evaluation.proto [hunk 1]
```diff
@@ -0,0 +1,200 @@
...
+}
+
+message EvaluationNamespaceSnapshotRequest {
+  string key = 1;
+}
+
+// flipt:sdk:ignore
+service DataService {
+  rpc EvaluationSnapshotNamespace(EvaluationNamespaceSnapshotRequest) returns (EvaluationNamespaceSnapshot);
+}
```
Comment (yquansah) [inline flipt-engine/proto/evaluation.proto:200]: This file could potentially be a moving target if things change in the original repository, just to point it out.

File: flipt-engine/src/models/common.rs [hunk 2]
```diff
@@ -9,6 +9,15 @@ pub enum FlagType {
     Boolean,
 }
 
+impl From<i32> for FlagType {
+    fn from(n: i32) -> FlagType {
+        match n {
+            1 => FlagType::Boolean,
+            _ => FlagType::Variant,
```
Comment (yquansah) [inline flipt-engine/src/models/common.rs:16]:
```suggestion
            0 => FlagType::Variant,
            _ => FlagType::Variant,
```

Maybe we should be all the way explicit here.

File: flipt-engine/src/models/common.rs [hunk 3]
```diff
@@ -18,6 +27,15 @@ pub enum SegmentOperator {
     And,
 }
 
+impl From<i32> for SegmentOperator {
+    fn from(n: i32) -> SegmentOperator {
+        match n {
+            1 => SegmentOperator::And,
+            _ => SegmentOperator::Or,
```
Comment (yquansah) [inline flipt-engine/src/models/common.rs:34]:
```suggestion
            0 => SegmentOperator::Or,
            _ => SegmentOperator::Or,
```

Same here, maybe we should be all the way explicit.

File: flipt-engine/src/models/common.rs [hunk 4]
```diff
@@ -26,6 +44,14 @@ pub enum SegmentMatchType {
     #[serde(rename = "ALL_SEGMENT_MATCH_TYPE")]
     All,
 }
+impl From<i32> for SegmentMatchType {
+    fn from(n: i32) -> SegmentMatchType {
+        match n {
+            1 => SegmentMatchType::Any,
+            _ => SegmentMatchType::All,
```
Comment (yquansah) [inline flipt-engine/src/models/common.rs:51]:
```suggestion
            0 => SegmentMatchType::All,
            _ => SegmentMatchType::All,
```

Same here.

File: flipt-engine/proto/evaluation.proto (hunk 1 above)
Comment (erka) [reply flipt-engine/proto/evaluation.proto:200]: Yep, ideally it would be to have this in `flipt-rust` client and use it in this project. What do you think about this @markphelps?

Comment (markphelps) [reply flipt-engine/proto/evaluation.proto:200]: yeah, we're actively refactoring our Server side SDKs at the moment (rust included), so I would hold off for now

Also the `DataService` is really meant to be 'internal' in that users shouldn't depend on its structure (although we will always aim for backward compatibility in reality because protos).

tl;dr I think this is fine for now

File: flipt-engine/examples/grps.rs [hunk 5]
```diff
@@ -0,0 +1,29 @@
+// cargo run --example grps
```
Comment (markphelps) [inline flipt-engine/examples/grps.rs:1]: should this file be named `grpc`?

File: flipt-engine/gen/Cargo.toml [hunk 6]
```diff
@@ -0,0 +1,14 @@
+[package]
+name = "protos"
```
Comment (markphelps) [inline flipt-engine/gen/Cargo.toml:2]: this doesnt get built into the `engine` lib correct? just want to ensure the engine libraries are as small as they can be since they are embedded in all the clients

**Review by markphelps — COMMENTED** [review]
great idea @erka !!! just a couple minor questions

File: flipt-engine/examples/grps.rs (hunk 5 above)
Comment (erka) [reply flipt-engine/examples/grps.rs]: You're right :)

File: flipt-engine/gen/Cargo.toml (hunk 6 above)
Comment (erka) [reply flipt-engine/gen/Cargo.toml:2]: With this goal we don't need `tonic`. I built release for main and this branches. It's results.
```
// main
768B libfliptengine.d
8.2M libfliptengine.dylib
1.5M libfliptengine.rlib

// this 
928B libfliptengine.d
8.3M libfliptengine.dylib
1.3M libfliptengine.rlib
```
